client/src/utils/hooks.js

```diff
@@ -1,14 +1,22 @@
 import { useState, useEffect } from 'react';
 
-export const useFetch = (url) => {
+export const useFetch = (url, setIsLoading) => {
```

Comment:
is this of setting loading legit or is there a better way?

Reply:
Probably shouldn't mixed the data fetching logic and ui logic together. I would separate them out.